### PR TITLE
M3-525

### PR DESCRIPTION
**Pull request by rjstires**
## Purpose
Volume creation is asynchronous and we needed a pattern for acknowledging the request was successful, and then notifying the user that the creation was successful.

We now display a message within the drawer that the volume was queued for creation. The drawer will then close automatically (after 4 seconds). When the volume has been created (and optionally attached) a toast notification displays alerting the user.

## Gold Plating
- Updated usage of Button to our abstraction.
- Added submitting state.
- Added composeState instance method.
- Added lenses for state composition.
- Added generic API request catch handler.

### CODE REVIEW

File: src/features/Volumes/VolumeDrawer.tsx [hunk 1]
```diff
@@ -204,31 +246,29 @@ class VolumeDrawer extends React.Component<CombinedProps, State> {
...
         if (!label) {
-          if (this.mounted) {
-            this.setState({
-              errors: [{ field: 'label', reason: 'Label cannot be blank.' }],
-            }, () => {
-              scrollErrorIntoView();
-            });
-          }
+          this.composeState([
+            set(L.errors, [{ filed: 'label', reason: 'Label cannot be blank.' }])
```
Comment (Jskobos) [inline src/features/Volumes/VolumeDrawer.tsx]: Field

File: src/features/Volumes/VolumeDrawer.tsx [hunk 2]
```diff
@@ -117,6 +132,16 @@ class VolumeDrawer extends React.Component<CombinedProps, State> {
     configs: [],
   };
 
+  handleAPIErrorResponse = (errorResponse: any) => this.composeState([
+    set(L.errors, path(['response', 'data', 'errors'], errorResponse))
+  ], () => scrollErrorIntoView());
```
Comment (Jskobos) [inline src/features/Volumes/VolumeDrawer.tsx]: this.state.submitting needs to be set to false, either as part of this method or elsewhere in the flow. As is, the button remains in loading state after returning an error, and only refreshing the page returns it to active.

File: src/features/Volumes/VolumeDrawer.tsx (hunk 1 above)
Comment (rjstires) [reply src/features/Volumes/VolumeDrawer.tsx]: Fixed, good eye.

File: src/features/Volumes/VolumeDrawer.tsx (hunk 2 above)
Comment (rjstires) [reply src/features/Volumes/VolumeDrawer.tsx]: Fixed, thank you.

**Comment by rjstires** [comment]
@alioso I've reset the form on close, good call. I think we need to develop and visual and technical pattern for doing as you've reqested.

**Comment by Jskobos** [comment]
Submitting etc. working well now. I think you could also clear the error state at the beginning `onSubmit` or in the response handler. It's still possible to get into a slightly weird state:

1. cause an error (error Notice is displayed)
2. fix the error and resubmit (success Notice is displayed, error is still there)

**Comment by rjstires** [comment]
@Jskobos That's a good call, we do that elsewhere. FWIW this isnt necessarily a result of this PR but I've gone ahead and addressed it anyhow.

**Comment by alioso** [comment]
@rjstires all set here, looking good. As discussed I added a "Create a Volume" button on the landing page

**Comment by patthiel** [comment]
Looks good to me. I pushed some e2e test changes that increase the timeouts to account for the drawer not closing instantly.

**Pull request merged**